Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -332,34 +351,68 @@ public void FailPlugin(Assembly assembly)
             ChangePluginState(plugin, PluginStatus.Malfunctioned);
         }
 
-        /// <summary>
-        /// Saves the manifest back to disk.
-        /// </summary>
-        /// <param name="manifest">The <see cref="PluginManifest"/> to save.</param>
-        /// <param name="path">The path where to save the manifest.</param>
-        /// <returns>True if successful.</returns>
+        /// <inheritdoc/>
         public bool SaveManifest(PluginManifest manifest, string path)
         {
-            if (manifest == null)
-            {
-                return false;
-            }
-
             try
             {
                 var data = JsonSerializer.Serialize(manifest, _jsonOptions);
                 File.WriteAllText(Path.Combine(path, "meta.json"), data);
                 return true;
             }
-#pragma warning disable CA1031 // Do not catch general exception types
-            catch (Exception e)
-#pragma warning restore CA1031 // Do not catch general exception types
+            catch (ArgumentException e)
             {
-                _logger.LogWarning(e, "Unable to save plugin manifest. {Path}", path);
+                _logger.LogWarning(e, "Unable to save plugin manifest due to invalid value. {Path}", path);
                 return false;
             }
         }
 
+        /// <inheritdoc/>
+        public async Task<bool> GenerateManifest(PackageInfo packageInfo, Version version, string path)
+        {
+            if (packageInfo == null)
+            {
+                return false;
+            }
+
+            var versionInfo = packageInfo.Versions.First(v => v.Version == version.ToString());
+            var imagePath = string.Empty;
+
+            if (!string.IsNullOrEmpty(packageInfo.ImageUrl))
+            {
+                var url = new Uri(packageInfo.ImageUrl);
+                imagePath = Path.Join(path, url.Segments.Last());
+
+                await using var fileStream = File.OpenWrite(imagePath);
+                var downloadStream = await HttpClientFactory
+                    .CreateClient(NamedClient.Default)
+                    .GetStreamAsync(url)
+                    .ConfigureAwait(false);
+
+                await downloadStream.CopyToAsync(fileStream).ConfigureAwait(false);
+                await fileStream.DisposeAsync();
```

Comment:
```suggestion
```

This is already being disposed, no need to manually call
(I know Rider suggests adding this, but it's not needed)